src/bool.rs

```diff
@@ -0,0 +1,132 @@
+//! Functions for safe transmutation to `bool`.
+//! Transmuting to `bool' is not undefined behavior if the transmuted value is either 0 or 1.
+//! These functions will return an error if the integer value behind the `bool` value is neither one.
+
+use self::super::{Error, ErrorReason, guarded_transmute_vec_permissive, guarded_transmute_many_permissive, guarded_transmute_many_pedantic,
+                  guarded_transmute_vec_pedantic};
+
+use std::mem::align_of;
+
+/// Makes sure that the bytes represent a sequence of valid boolean values. It is done
+/// this way because the language does not guarantee that `bool` is 1 byte sized.
+#[cfg(target_endian = "little")]
+#[inline]
+fn bytes_are_bool(v: &[u8]) -> bool {
+    let sizeof_bool: usize = align_of::<bool>();
+    for c in v.chunks(sizeof_bool).filter(|c| c.len() == sizeof_bool) {
+        if c[0] > 1 || c[1..].iter().any(|x| *x != 0) {
+            return false;
+        }
+    }
+    true
+}
+
+/// Makes sure that the bytes represent a sequence of valid boolean values. It is done
+/// this way because the language does not guarantee that `bool` is 1 byte sized.
+#[cfg(target_endian = "big")]
+#[inline]
+fn bytes_are_bool(v: &[u8]) -> bool {
+    let sizeof_bool: usize = align_of::<bool>();
+    for c in v.chunks(sizeof_bool).filter(|c| c.len() == sizeof_bool)  {
+        let (l, r) = c.split_at(sizeof_bool - 1);
```

Comment:
The entirety of these functions is the same, apart from picking {L,M}SB. I feel like this would work as well, and better express what's happening:

```rust
let (msbs, lsb) = if cfg!(target_endian = "little") {
    (c[1..], c)
} else {
    c.split_at(sizeof_bool - 1)
};
if lsb[0] > 1 || msbs.iter().any(|x| *x != 0) {
```

Also, you have an unhealthy obsession with naming verbols for that they are rather than for what they represent `(l, r)` mean naught, `(l, m)` or `(lsb, msb)` or sth would be :+1:er.

Reply:
Ok, I took care of that. Glad the compiler's smart enough to optimize the if-else statement.

That last bit's subjective, but fine. I improved the variable naming as well.